### ISSUE TITLE
Stress Tests

### ISSUE THREAD

**Issue by vasco-santos** [ChainSafe/gossipsub-js#23]
Before deploying `gossipsub` into the wild in the `libp2p` network, it would be great to have a suite of stress tests

**Comment by Mikerah** [comment]
What would this entail? We should perhaps have a list of what we want to stress test and the environment parameters. 

Did the `js-libp2p` team have stress tests for `floodsub`?

**Comment by vasco-santos** [comment]
I checked the remaining issues and found this one [ChainSafe/gossipsub-js#10](https://github.com/ChainSafe/gossipsub-js/issues/10). They both fall in the same category and can be merged in a single issue.

I expect with stress tests to launch a large set of peers and do an intensive amount of work with `publish` and `subscribe` actions, and guarantee that the expected data is received by each peer. There are no stress tests for `floodsub` as it was implemented a while ago and it is behind an experimental flag in `js-libp2p`. The most recent work in `js-libp2p` has adopted the good practise of benchmarking, such as [libp2p/pull-mplex/tree/master/benchmarks](https://github.com/libp2p/pull-mplex/tree/master/benchmarks).

We do not need to have a really complete set of tests regarding benchmarks, but I think it would be important to have a minimum guarantee of scalability before adding `gossipsub` in `js-libp2p` without the experimental flag. We can iterate those tests after
